Wikipedia/Code/NSURL+WMFLinkParsing.m

```diff
@@ -180,7 +180,13 @@ + (NSURL *)wmf_desktopURLForURL:(NSURL *)url {
 #pragma mark - Properties
 
 - (BOOL)wmf_isWikiResource {
-    return [self.path wmf_isWikiResource];
+    static NSString *wikiResourceSuffix = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        wikiResourceSuffix = [NSString stringWithFormat:@".%@", WMFDefaultSiteDomain];
+    });
+    NSString *lowercaseHost = self.host.lowercaseString;
+    return (!lowercaseHost || [lowercaseHost isEqualToString:WMFDefaultSiteDomain] || [lowercaseHost hasSuffix:wikiResourceSuffix]) && [self.path wmf_isWikiResource];
```

Comment:
Or wait… never mind…

If we check for  "wikipedia.org", we also need to check for other "known" domains. For instance this breaks for "mediawiki.org"